### PR TITLE
Single client

### PR DESCRIPTION
**Pull request by andrewMacmurray**
+ Can start an echo server
+ Can connect to server with echo client
+ Server will echo back messages sent to it via the client

### CODE REVIEW

File: src/test/java/EchoServerTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        outputStream = new ByteArrayOutputStream();
+
+        mockClientSocket = mock(Socket.class);
+        mockServerSocket = mock(ServerSocket.class);
+
+        configureMocks();
+    }
+
+    private void configureMocks() throws IOException {
+        // clientSocket
```
Comment (sarahabimay) [inline src/test/java/EchoServerTest.java]: unnecessary comment, no?

File: src/test/java/EchoServerTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        when(mockClientSocket.getInputStream()).thenReturn(inputStream);
+        when(mockClientSocket.getOutputStream()).thenReturn(outputStream);
+
+        // serverSocket
```
Comment (sarahabimay) [inline src/test/java/EchoServerTest.java]: same thing

File: src/main/java/client/MyClient.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package client;
+
+import client.EchoClient;
+
```
Comment (sarahabimay) [inline src/main/java/client/MyClient.java]: why this space between imports in all of the files?  I don't think it adds anything.

**Review by sarahabimay — COMMENTED** [review]
Looks really good.  
You should try experimenting without Mockito since it is more challenging not to use it, and you're obviously already comfortable with the language and design aspects.

File: src/test/java/EchoServerTest.java (hunk 1 above)
Comment (andrewMacmurray) [reply src/test/java/EchoServerTest.java]: good point, I think it was a lot more complicated before and forgot to take it out!

**Comment by andrewMacmurray** [comment]
@sarahabimay thanks! I'll have a go without Mockito

File: src/main/java/client/EchoClient.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import java.net.Socket;
+
+public class EchoClient {
+
+    private SocketIO socketIO;
+    private StdIO stdIO;
+
+    public EchoClient(Socket socket, InputStream in, PrintStream out) throws IOException {
+        socketIO = new SocketIO(socket);
+        stdIO = new StdIO(in, out);
```
Comment (matthewglover) [inline src/main/java/client/EchoClient.java]: Could you just pass in the StdIO?

**Pull request merged**